Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -467,7 +476,8 @@ private func createResolvedPackages(
             for case .product(let productRef, let conditions) in targetBuilder.target.dependencies {
                 // Find the product in this package's dependency products.
                 // Look it up by ID if module aliasing is used, otherwise by name.
-                let product = lookupByProductIDs ? productDependencyMap[productRef.identity] : productDependencyMap[productRef.name]
+                let key = "\(productRef.package?.lowercased() ?? productRef.name.lowercased())_\(productRef.name)"
```

Comment:
Potentially this should really be the value of `productRef.identity`?

Reply:
seems sensible to compute this at the product level and use everywhere

Reply:
It looks like `productRef.identity` is used by `ModuleAliasTracker` as well, so in order to keep this fix small, I am not going to change it for now.

Reply:
Actually, it didn't even need to change, this was already what `productRef.identity` returned.